src/components/Users/UserResetPassword.tsx

```diff
@@ -1,207 +1,217 @@
-import request from "@/Utils/request/request";
+import mutate from "@/Utils/request/mutate";
 import { UserBase } from "@/types/user/user";
 
-import ButtonV2 from "../Common/ButtonV2";
-
-interface PasswordForm {
-  username: string;
-  old_password: string;
-  new_password_1: string;
-  new_password_2: string;
-}
+const PasswordSchema = z
+  .object({
+    old_password: z
+      .string()
+      .min(1, { message: t("please_enter_current_password") }),
+    new_password_1: z
+      .string()
+      .min(8, { message: t("password_length_validation") })
+      .regex(/\d/, { message: t("password_number_validation") })
+      .regex(/[a-z]/, {
+        message: t("password_lowercase_validation"),
+      })
+      .regex(/[A-Z]/, {
+        message: t("password_uppercase_validation"),
+      }),
+    new_password_2: z
+      .string()
+      .min(1, { message: t("please_enter_confirm_password") }),
+  })
+  .refine((values) => values.new_password_1 === values.new_password_2, {
+    message: t("password_mismatch"),
+    path: ["new_password_2"],
+  })
+  .refine((values) => values.new_password_1 !== values.old_password, {
+    message: t("new_password_same_as_old"),
+    path: ["new_password_1"],
+  });
 
 export default function UserResetPassword({
   userData,
 }: {
   userData: UserBase;
 }) {
   const { t } = useTranslation();
-  const [isSubmitting, setisSubmitting] = useState(false);
   const [isEditing, setIsEditing] = useState(false);
-
-  const initForm: PasswordForm = {
-    username: userData.username,
-    old_password: "",
-    new_password_1: "",
-    new_password_2: "",
-  };
-
-  const validateNewPassword = (password: string) => {
-    if (
-      password.length < 8 ||
-      !/\d/.test(password) ||
-      password === password.toUpperCase() ||
-      password === password.toLowerCase()
-    ) {
-      return false;
-    }
-    return true;
-  };
-
-  const validateForm = (formData: PasswordForm) => {
-    const errors: Partial<Record<keyof PasswordForm, string>> = {};
-
-    if (!formData.old_password) {
-      errors.old_password = t("please_enter_current_password");
-    }
-
-    if (!formData.new_password_1) {
-      errors.new_password_1 = t("please_enter_new_password");
-    } else if (!validateNewPassword(formData.new_password_1)) {
-      errors.new_password_1 = t("new_password_validation");
-    }
-
-    if (!formData.new_password_2) {
-      errors.new_password_2 = t("please_confirm_password");
-    } else if (formData.new_password_1 !== formData.new_password_2) {
-      errors.new_password_2 = t("password_mismatch");
-    }
-
-    if (formData.new_password_1 === formData.old_password) {
-      errors.new_password_1 = t("new_password_same_as_old");
-    }
-
-    return errors;
-  };
-
-  const handleSubmit = async (formData: PasswordForm) => {
-    setisSubmitting(true);
+  const [isPasswordFieldFocused, setIsPasswordFieldFocused] = useState(false);
+
+  const resetPasswordForm = useForm({
```

Comment:
the file is anyways named `UserResetPassword.tsx`, let's keep the names simple since there's no other forms in the same file.

```suggestion
  const form = useForm({
```